frontend/src/app/eight-counter/eight-counter.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, Input, OnChanges, OnInit, Output, SimpleChanges } from '@angular/core';
+
+@Component({
+  selector: 'app-eight-counter',
+  templateUrl: './eight-counter.component.html',
+  styleUrls: ['./eight-counter.component.scss']
+})
+export class EightCounterComponent implements OnChanges, OnInit {
+
+  @Input()
+  duration: number;
+  @Input()
+  tempo: number;
+  @Input()
+  frameIndex: number;
+  @Input()
+  isPlayingOn: boolean;
+  @Input()
+  isVoiceSynthesisOn: boolean;
+
+  @Output()
+  synthesisVoiceOptions: string[];
+
+  counterNumber: number;
+  speechSynthesisWindow: SpeechSynthesis;
+  speechSynthesis: SpeechSynthesisUtterance;
+  timerId: number;
+  progressBarValue = 0;
+  timeElapsedInFrame = 0;
+
+  constructor() {
+  }
+
+  ngOnInit(): void {
+    this.speechSynthesisWindow = window.speechSynthesis;
+    this.speechSynthesis = new SpeechSynthesisUtterance();
+    this.speechSynthesis.rate = 2;
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    this.setCounterNumber();
```

Comment:
**suggestion:** add an if statement for checking which changes are included. I.e. `if (changes.frame) { ... }`